sphinx_immaterial/apidoc/cpp/api_parser.py

```diff
@@ -429,6 +441,41 @@ def _get_all_decls(config: Config, cursor: Cursor, allow_file):
             yield from _get_all_decls(config, child, None)
 
 
+def strip_comment(cmt: str = None) -> List[str]:
+    """Strip the raw string of an object's comment into lines.
+    :param cmt: the comment to parse.
+    :returns: A list of the lines without the surrounding C++ comment syntax.
+    """
+    if cmt is None:
+        return [""]
+    # the first line is never indented (in clang-parsed form)
+    # so dedent all subsequent lines only
+    first_line_end = cmt.find("\n")
+    cmt = cmt[:first_line_end] + dedent(cmt[first_line_end:])
+    # split into a list of lines & account for CRLF and LF line endings
+    body = [line.rstrip("\r") for line in cmt.splitlines()]
+
+    # strip all the comment syntax out
+    if body[0].startswith("//"):
+        body = [line.lstrip("/").lstrip("!").lstrip("<") for line in body]
+    elif body[0].startswith("/*"):
+        body[0] = body[0].lstrip("/").lstrip("*").lstrip("!")
+        multi_lined_asterisk = True  # works also for single-line comments blocks
+        if len(body) > 1:
+            line_has_asterisk = [line.startswith("*") for line in body[1:]]
+            multi_lined_asterisk = line_has_asterisk.count(True) == len(body) - 1
+        body = [
+            (line.lstrip("*").lstrip("<") if multi_lined_asterisk else line)
+            for line in body
+        ]
+        body[-1] = body[-1].rstrip("*/").rstrip()
+    body = dedent("\n".join(body)).splitlines()
```

Comment:
>  But I think we should only dedent multiline comments.

I added a condition to skip dedenting single line comments, but the doc builds fail because the leading whitespace is now being interpreted as content to the ext's generated `.. highlight:: cpp` directive.

> ```
> <cpp_apigen_rst_prolog>:6: ERROR: Error in "highlight" directive:
> no content permitted.
> .. highlight:: cpp
>  Returns the data order.
> ```

Kinda glad I did the rebase on main now (this would've gone unnoticed until much later).

Reply:
To be clear, A single line comment (with respect to `raw_comment`) is a block that has 1 line

```cpp
/// A single line comment

/// Not a single line comment because
/// it has multiple lines in sequence
```
So, I'm not sure what the advantage is for not dedenting a single line comment. Maybe we could delegate this to regex with `re.sub(r"^(\s{0, 3})\S", "", text)` (notice there's no multiline flag). Any other concerns I can think of all require multiple lines (eg. lists)